### PR TITLE
fix(batch): make `instanceType` optional on multinode jobs

### LINKED ISSUE #27185: aws-batch: MultiNodeJobDefinition Incorrectly Requires InstanceType
### Describe the bug

It seems that the Batch [MultiNodeJobDefinition CDK Construct](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_batch.MultiNodeJobDefinition.html) requires instance type to be defined. However, instanceType is not required by the Batch AWS SDK.

This seems to be more than a documentation issue, as I'm unable to create a MultiNodeJobDefinition in my CDK package without specifying instance type.

This constraint takes away from one of the primary advantages to having Batch manage our EC2 resources in that they optimally assign batch jobs to instance types based on availability/resource requirements.

### Expected Behavior

We shouldn't need to specify instance type. Batch should determine it based on job resource requirements.

### Current Behavior

CDK requires instance type to be required at job definition level.

### Reproduction Steps

Create a MultiNodeJobDefinition.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.96.2

### Framework Version

_No response_

### Node.js Version

18.x

### OS

linux

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by comcalvi**
Make `instanceType` optional. It used to be required. It will default to a batch optimal instance type.

The breaking change is for `MultiNodeJobDefinition.instanceType`, which is now optional. This is not truly breaking because everyone has this property explicitly set now, and will continue to have it set after this change. 

Closes #27185.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by TheRealAmazonKendra** [comment]
Weird that setting something to optional is considered a breaking change.

File: allowed-breaking-changes.txt [hunk 1]
```diff
@@ -188,3 +188,6 @@ removed:aws-cdk-lib.aws_cloudwatch.CfnAlarm.attrId
 removed:aws-cdk-lib.aws_ec2.CfnNetworkInterface.enablePrimaryIpv6
 removed:aws-cdk-lib.aws_ec2.CfnNetworkInterfaceAttachment.attrId
 removed:aws-cdk-lib.aws_ec2.CfnNetworkInterfaceProps.enablePrimaryIpv6
+
+# Make this optional so Batch can select the optimal instance type by default
+changed-type:aws-cdk-lib.aws_batch.MultiNodeJobDefinition.instanceType
```
Comment (rix0rrr) [inline allowed-breaking-changes.txt:193]: Why is it okay to make breaking changes to a stable API here? Especially since this  can be avoided?

**Comment by rix0rrr** [comment]
On the question of making properties optional, this comes up way too often, so I made an explainer for it:

https://github.com/aws/jsii/pull/4274/files?short_path=f5d91d1#diff-f5d91d104bff5f6eba3495d9aa4ef6d0e2854a010d6c2754acdd98536afc3c05

**Comment by comcalvi** [comment]
thanks for writeup @rix0rrr, that's helpful!

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**